File: valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) return false;
        
        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
        
        char[] sArray = s.toCharArray();
        char[] tArray = t.toCharArray();
        
        for(Character c : sArray){
            int count = map.getOrDefault(c, 0);
            map.put(c, count + 1);
        }
        
        for(Character c : tArray){
            if(map.containsKey(c)){
                int count = map.get(c);
                count--;
                if(count < 0) return false;
                map.put(c, count);
            }
            else{
                return false;
            }
           
        }
        
        
        return true;
    }
}

File: valid-anagram/valid-anagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        
        if(s.length != t.length) return false
        
        val map = HashMap<Char, Int>()
        
        val sArray = s.toCharArray()
        val tArray = t.toCharArray()
        
        for(c in sArray) map[c] = map.getOrDefault(c, 0) + 1
        
        for(c in tArray){
            var current = map.getOrDefault(c, 0)
            if(current < 1) return false
            map[c] = --current
        }
        
        return true
        
    }
}